src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -960,6 +1038,80 @@ sol_http_server_send_response(struct sol_http_request *request, struct sol_http_
     return ret;
 }
 
+SOL_API struct sol_http_response_progressive *
+sol_http_server_send_response_progressive(struct sol_http_request *request,
+    const struct sol_http_response *response,
+    void (*on_del)(void *data, const struct sol_http_response_progressive *progressive),
+    const void *cb_data)
+{
+    int ret;
+    struct sol_http_response_progressive *progressive;
+    struct MHD_Response *mhd_response;
+
+    SOL_NULL_CHECK(request, NULL);
+    SOL_NULL_CHECK(request->connection, NULL);
+    SOL_NULL_CHECK(response, NULL);
+
+    progressive = calloc(1, sizeof(*progressive));
+    SOL_NULL_CHECK(progressive, NULL);
+
+    sol_buffer_init(&progressive->buffer);
+    progressive->connection = request->connection;
+    progressive->delete_me = false;
+
+    MHD_resume_connection(request->connection);
+
+    mhd_response = build_mhd_response_progressive(response, progressive);
+    SOL_NULL_CHECK_GOTO(mhd_response, err);
+
+    ret = MHD_queue_response(request->connection, SOL_HTTP_STATUS_OK, mhd_response);
+    MHD_destroy_response(mhd_response);
+
+    SOL_INT_CHECK_GOTO(ret, != MHD_YES, err);
+
+    progressive->on_del = on_del;
+    progressive->cb_data = cb_data;
+
+    return progressive;
+
+err:
+    free(progressive);
+    return NULL;
+}
+
+SOL_API void
+sol_http_response_progressive_del(struct sol_http_response_progressive *progressive, bool graceful_del)
+{
+    SOL_NULL_CHECK(progressive);
+    SOL_EXP_CHECK(progressive->delete_me == true);
+
+    if (progressive->suspended)
+        MHD_resume_connection(progressive->connection);
+
+    progressive->graceful_del = graceful_del;
+    progressive->delete_me = true;
```

Comment:
IMO it's safer to always set these before calling other libraries. For some reason if `MHD_resume_connection()` calls you directly, you'd miss these values.